docs/4_dynamic.md

```diff
@@ -0,0 +1,166 @@
+# Dynamic Provisioning
+
+While someone in your organization needs to be familiar with the lowest level of
+infrastructure, many people just want a simple workflow for acquiring and using
+infrastructure. Crossplane provides the ability for organizations to define a
+*service catalog* of infrastructure and applications, then for teams and
+individuals to consume from the catalog with abstract requests. This is made
+possible by *dynamic provisioning*.
+
+> **Service Catalog**: a collection of services that is curated and defined by
+> infrastructure owners for usage across teams in an organization.
```

Comment:
Did we decide to introduce the "service catalog" concept to Crossplane? I recall it being discussed a few times, but I can't find any evidence of it in our various terminology documents. I'm a little wary of introducing it given that https://kubernetes.io/docs/concepts/extend-kubernetes/service-catalog/ is already a thing, and given that we don't actually refer to apps or infrastructure as "services" in any other context.

Reply:
Not sure, seemed like an opportune place to talk about it, but I would defer to @prasek here